### ISSUE TITLE
Head of `develop` doesn't currently compile

### ISSUE THREAD

**Issue by jcfreeman2** [DUNE-DAQ/hdf5libs#70]
Currently the code at the head of the `develop` branch isn't compiling against the current nightly (details in last night's [single repo CI build](https://github.com/DUNE-DAQ/hdf5libs/actions/runs/7752512672/job/21142147567)). Long story short, it appears that a function called `get_component_for_source_id` is called on a `TriggerRecordHeader` object, but this function doesn't exist.

**Comment by bieryAtFnal** [comment]
Hmm, I attempted to merge the addition of the `get_component_for_source_id` change to the `develop` branch in the `daqdataformats` repo [here](https://github.com/DUNE-DAQ/daqdataformats/pull/50).  Is there something additional that I need to do in `daqdataformats`?

**Comment by jcfreeman2** [comment]
On further reflection the "missing" `get_component_for_source_id` is a red herring since the nightly build failed and hence the single repo CI was run off the Feb-1 nightly - before the update to `daqdataformats`. 

However: I see that a dependency on `trgdataformats` has been added to `hd5libs`, so the reason the nightly failed is that a corresponding dependency needed to be added to `hdf5libs`' Spack `package.py` file in daq-release. I've added this dependency and have restarted the nightly.
